### PR TITLE
Fix adding an option with label '0'

### PR DESCRIPTION
**Pull request by mbijnsdorp**
### Description (*)
When adding an option to an attribute programmatically via `\Magento\Eav\Model\Entity\Attribute\OptionManagement::add`, it's not possible to use the label '0' for the option.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mbijnsdorp** [comment]
@magento run all tests

**Comment by mbijnsdorp** [comment]
@magento run Functional Tests B2B

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @mbijnsdorp,
Your changes look good to me.
Could you cover your changes with some automated tests?

**Comment by mbijnsdorp** [comment]
Hi @ihor-sviziev,
I'm inexperienced when it comes to writing tests. I could adjust the label of the add method https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php#L68, but that wouldn't make it very obvious. Would you suggest a similar method to that one?

**Comment by eduard13** [comment]
Hi @mbijnsdorp, thank you for your contribution. The existing test you mentioned, is a good place to cover your case. I'd suggest adding a simple dataProvider, that will provide different options labels to the `add` method. Something like in the following Unit Test.

https://github.com/magento/magento2/blob/f3e8f25987c7748d432cbe9d0fa2c54ff2a097bb/app/code/Magento/Customer/Test/Unit/Block/Widget/NameTest.php#L179-L211

Please let me know if you have any questions.

Thank you.

**Comment by mbijnsdorp** [comment]
@magento run all tests

**Comment by mbijnsdorp** [comment]
@magento run Unit Tests

**Comment by mbijnsdorp** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE, Functional Tests EE

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @eduard13 @mbijnsdorp,
Could you cover the case with the "update" method?

**Comment by mbijnsdorp** [comment]
Hi @ihor-sviziev, I see there's no test for that method at all yet. I'll try to get started on it later today

**Comment by mbijnsdorp** [comment]
@magento run Unit Tests

**Comment by mbijnsdorp** [comment]
@magento run all tests

File: app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php [hunk 1]
```diff
@@ -60,8 +60,8 @@ public function add($entityType, $attributeCode, $option)
     {
         $attribute = $this->loadAttribute($entityType, (string)$attributeCode);
 
-        $label = trim($option->getLabel() ?: '');
-        if (empty($label)) {
+        $label = trim((string) $option->getLabel());
```
Comment (ihor-sviziev) [inline app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php]:
```suggestion
        $label = trim((string)$option->getLabel());
```

File: app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php [hunk 2]
```diff
@@ -93,8 +93,8 @@ public function update(
         if (empty($optionId)) {
             throw new InputException(__('The option id is empty. Enter the value and try again.'));
         }
-        $label = trim($option->getLabel() ?: '');
-        if (empty($label)) {
+        $label = trim((string) $option->getLabel());
```
Comment (ihor-sviziev) [inline app/code/Magento/Eav/Model/Entity/Attribute/OptionManagement.php]:
```suggestion
        $label = trim((string)$option->getLabel());
```

File: app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php [hunk 3]
```diff
@@ -59,13 +60,14 @@ protected function setUp(): void
 
     /**
      * Test to add attribute option
+     *
+     * @dataProvider optionLabelDataProvider
      */
-    public function testAdd()
+    public function testAdd($label)
```
Comment (ihor-sviziev) [inline app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php]:
```suggestion
    public function testAdd(string $label): void
```

File: app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php [hunk 4]
```diff
@@ -59,13 +60,14 @@ protected function setUp(): void
 
     /**
      * Test to add attribute option
+     *
+     * @dataProvider optionLabelDataProvider
```
Comment (ihor-sviziev) [inline app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php:65]:
```suggestion
     * @param string $label
     * @dataProvider optionLabelDataProvider
```

File: app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php [hunk 5]
```diff
@@ -216,6 +229,74 @@ public function testAddWithCannotSaveException()
         $this->model->add($entityType, $attributeCode, $optionMock);
     }
 
+    /**
+     * Test to update attribute option
+     *
+     * @dataProvider optionLabelDataProvider
+     */
+    public function testUpdate($label)
```
Comment (ihor-sviziev) [inline app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php]:
```suggestion
    public function testUpdate(string $label): void
```

File: app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php [hunk 6]
```diff
@@ -216,6 +229,74 @@ public function testAddWithCannotSaveException()
         $this->model->add($entityType, $attributeCode, $optionMock);
     }
 
+    /**
+     * Test to update attribute option
+     *
+     * @dataProvider optionLabelDataProvider
```
Comment (ihor-sviziev) [inline app/code/Magento/Eav/Test/Unit/Model/Entity/Attribute/OptionManagementTest.php:237]:
```suggestion
     * @param string $label
     * @dataProvider optionLabelDataProvider
```

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mbijnsdorp** [comment]
@ihor-sviziev Could you also add the partner Youwe label?

**Comment by ihor-sviziev** [comment]
@sidolov @sivaschenko, could you check why the partner label wasn't applied?

**Review by ihor-sviziev — APPROVED** [review]
@mbijnsdorp, great job! Thank you for your contribution!
✔ Approved

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9343](https://jira.corp.magento.com/browse/ENGCOM-9343) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@mbijnsdorp thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by sidolov** [comment]
@ihor-sviziev label wasn't added because @mbijnsdorp is not a part of Youwe team on the GitHub

**Comment by engcom-Dash** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Add the code to add product attribute manually with label ‘0’ (screenshot below) 

Screenshot 2022-01-12 at 12 41 04 PM

2. Run the code on the magento server 
 
**Before:** :heavy_multiplication_x:   Exception is throwing with message ‘The attribute option label is empty. Enter the value and try again.’

Screenshot 2022-01-12 at 12 44 01 PM


**After:** :heavy_check_mark:  The code run successfully without expection.

**Pull request merged**